### PR TITLE
Emplaced Turret Deployment Mechanics

### PR DESCRIPTION
**Pull request by mazzinsanity**
## About The Pull Request
Emplaced turrets (currently only the M2 Browning HMG) now have to be first wrenched and then welded to anchor and use them. In order to de-anchor the turret, you need to unweld and unwrench it, letting you move it.

## Why It's Good For The Game
As it currently stands, you can only anchor the M2 Browning once and then it becomes permanently fixed to its location until it is destroyed, which limits its use and is generally unfun to deal with.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
add: Added a way to move emplaced turrets after they have been anchored
/:cl:

### CODE REVIEW

File: code/game/objects/structures/manned_turret.dm [hunk 1]
```diff
@@ -27,6 +27,56 @@
...
+	else if(anchored)	//We have wrenched it already
+		to_chat(user, "<span class='notice'>You start unsecuring [src]...</span>")
+		if(I.use_tool(src, user, 4 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>You unsecure [src].</span>")
+			anchored = FALSE
+
+	else				//We haven't wrenched it yet
+		to_chat(user, "<span class='notice'>You start securing [src]...</span>")
+		if(I.use_tool(src, user, 4 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>You secure [src].</span>")
```
Comment (Draganfrukts) [inline code/game/objects/structures/manned_turret.dm:54]:
```suggestion
			to_chat(user, "<span class='notice'>You secure [src], time to make them eat lead!</span>")
```

File: code/game/objects/structures/manned_turret.dm [hunk 2]
```diff
@@ -27,6 +27,56 @@
...
+								"<span class='notice'>You unweld [src] from the floor.</span>")
+			unbuckle_all_mobs(TRUE)
+			can_buckle = FALSE
+			playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+
+	else if(anchored)	//We have wrenched it already
+		to_chat(user, "<span class='notice'>You start welding [src] to the floor...</span>")
+		if(I.use_tool(src, user, 4 SECONDS, volume=50, amount=2))
+			user.visible_message("<span class='warning'>[user] welds [src] to the floor.</span>",\
+								"<span class='notice'>You weld [src] to the floor.</span>")
```
Comment (Draganfrukts) [inline code/game/objects/structures/manned_turret.dm:76]:
```suggestion
								"<span class='notice'>You weld [src] to the floor, the party starts here!</span>")
```

File: code/game/objects/structures/manned_turret.dm [hunk 3]
```diff
@@ -27,6 +27,56 @@
...
 	return ..()
 
+/obj/machinery/manned_turret/examine(mob/user)
+	. = ..()
+	if(!anchored)	//We haven't wrenched it yet
+		. += "<span class='notice'>The bolts are <i>loosened</i>, allowing it to be moved.</span>"
+	else if(!can_buckle)	//We have wrenched it but haven't welded it yet
+		. += "<span class='notice'>The bolts are <b>secured</b>, but the anchor must be <i>welded</i> to the floor before use.</span>"
+	else	//We have wrenched and welded it, it's ready to be used
+		. += "<span class='notice'>The anchor is <b>welded</b> to the floor and the gun is ready to be used.</span>"
```
Comment (Draganfrukts) [inline code/game/objects/structures/manned_turret.dm:37]:
```suggestion
		. += "<span class='notice'>The anchor is <b>welded</b> to the floor and the gun is ready to rip.</span>"
```

**Review by Draganfrukts — COMMENTED** [review]
imperative to gameplay

**Review by Stutternov — APPROVED** [review]

**Pull request merged**